framework/src/Volo.Abp.Cli.Core/Volo/Abp/Cli/ProjectModification/PackageSourceSwitcher.cs

```diff
@@ -0,0 +1,179 @@
+            Logger.LogInformation($"Packages on solution \"{Path.GetFileName(solutionPath)}\" are being switched to {target}.");
+            Logger.LogInformation("");
+
+            var projectFiles = GetCsprojFiles(solutionPath);
+
+            foreach (var projectFile in projectFiles)
+            {
+                Logger.LogInformation($"Packages on project \"{Path.GetFileName(projectFile)}\" are being switched to {target}.");
+
+                var content = File.ReadAllText(projectFile);
+
+                var doc = new XmlDocument() { PreserveWhitespace = true };
+
+                doc.Load(GenerateStreamFromString(content));
+
+                doc = SwitchPackagesInDocument(doc, content, target);
+
+                File.WriteAllText(projectFile, doc.OuterXml);
+            }
+
+
+            Logger.LogInformation("");
+            Logger.LogInformation($"Packages on solution \"{Path.GetFileName(solutionPath)}\" are switched to {target}.");
+        }
+
+        private XmlDocument SwitchPackagesInDocument(XmlDocument doc, string content, SwitchTarget target)
+        {
+            Check.NotNull(content, nameof(content));
+
+            var nodes = doc.SelectNodes("/Project/ItemGroup/PackageReference[@Include]");
+
+            foreach (XmlNode node in nodes)
+            {
+                var packageName = node.Attributes["Include"].Value;
+
+                if (!packageName.StartsWith("Volo.Abp"))
+                {
+                    continue;
+                }
+
+                var version = node.Attributes["Version"].Value;
+
+                XmlNode newNode = GetNewReferenceNode(doc, packageName, version, target);
+
+                node.ParentNode.ReplaceChild(newNode, node);
+            }
+
+            return doc;
+        }
+
+        private  XmlElement GetNewReferenceNode(XmlDocument doc, string packageName, string version, SwitchTarget target)
+        {
+            var newNode = doc.CreateElement("PackageReference");
+            var newVersion = "";
+
+            if (target == SwitchTarget.Stable)
+            {
+                var versonSplitted = version.Split("-");
+
+                newVersion = versonSplitted.Length >= 2 ?
+                    version.Split("-")[0] : version;
+            }
+            if (target == SwitchTarget.Preview)
+            {
+                var versonSplitted = version.Split("-");
+
+                newVersion = versonSplitted.Length < 2 ?
+                    version + $"-preview{GetFormattedDate()}" : version;
+            }
+
+            var includeAttr = doc.CreateAttribute("Include");
+            includeAttr.Value = packageName;
+            newNode.Attributes.Append(includeAttr);
+
+            var versionAttr = doc.CreateAttribute("Version");
+            versionAttr.Value = newVersion;
+            newNode.Attributes.Append(versionAttr);
+            return newNode;
+        }
+
+        private string GetFormattedDate()
+        {
+            var formattedDate = "";
```

Comment:
Is there any specific reason to have a method like this? AFAIK, `DateTime.Now.ToString("yyyyMMdd")` should be enough to get an output like this `20200221`